### PR TITLE
feat/Python Dependency Dir and Download Dependency

### PR DESCRIPTION
**Pull request by mingkun2020**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws_lambda_builders/workflows/python_pip/workflow.py [hunk 1]
```diff
@@ -73,17 +73,28 @@ def __init__(self, source_dir, artifacts_dir, scratch_dir, manifest_path, runtim
...
+                        runtime,
+                        self.dependencies_dir,
+                        binaries=self.binaries,
+                    )
+                )
+            # if dependencies folder is provided, copy dependencies from dependencies folder to build folder
+            if self.dependencies_dir:
+                self.actions.append(
+                    CopySourceAction(self.dependencies_dir, artifacts_dir, excludes=self.EXCLUDED_FILES)
+                )
```
Comment (mndeveci) [inline aws_lambda_builders/workflows/python_pip/workflow.py:94]: Same comment from the other PR (Ruby, NodeJS). What happens if `self.dependencies_dir` is not defined?

File: aws_lambda_builders/workflows/python_pip/workflow.py [hunk 2]
```diff
@@ -92,8 +92,16 @@ def __init__(self, source_dir, artifacts_dir, scratch_dir, manifest_path, runtim
...
                 )
+
         else:
             LOG.warning("requirements.txt file not found. Continuing the build without dependencies.")
+
+        if not self.download_dependencies and not self.dependencies_dir:
+            LOG.info(
+                "download_dependencies is false and the dependencies_dir is not exists, just copying the source files "
+                "into the artifacts directory "
+            )
```
Comment (mndeveci) [inline aws_lambda_builders/workflows/python_pip/workflow.py:103]: I would move validation to the above so that we will fail or emit this message early.

**Review by mndeveci — APPROVED** [review]
LGTM, just like the other one, both options (logging vs raising an exception) is fine by me.

**Review by CoshUS — APPROVED** [review]
LGTM!

**Pull request merged**